### PR TITLE
Popular diaper option, refactor

### PR DESCRIPTION
**Pull request by apkatsikas**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -230,48 +232,58 @@ async function pickADiaper (msg) {
...
+      // Split on tabs
+      diapersInMemory.push(line.split('\t'));
+      // Only get semi-popular people
+      const popularViews = 5000000;
+      diapersInMemory = diapersInMemory.filter((item) => { return item[pageViewsIx] >= popularViews; });
+    }, (err) => {
+      if (err) { return reject(err); };
+
+      // Return random person
+      const randomInt = generateRandomInt(diapersInMemory.length);
```
Comment (rmag42) [inline app.js]: diapersInMemory.length - 1. This is index, starting at 0.

File: app.js [hunk 2]
```diff
@@ -230,48 +232,58 @@ async function pickADiaper (msg) {
...
-      if (i === randomInt) {
-        resolve(line);
-        return false;
-      }
-      i++;
+    lineReader.eachLine(diaperFileName, (line) => {
+      // Split on tabs
+      diapersInMemory.push(line.split('\t'));
+      // Only get semi-popular people
+      const popularViews = 5000000;
```
Comment (rmag42) [inline app.js:268]: Might give this global scope like the pageViewsIx, in case we want to change the number

File: app.js (hunk 1 above)
Comment (rmag42) [reply app.js]: Actually minus two because of the header.

Comment (apkatsikas) [reply app.js]: it looks like youre reviewing an outdated version, try refresh? i have the header and the -1 in my version but it was commited later

File: app.js (hunk 2 above)
Comment (apkatsikas) [reply app.js]: done

File: app.js [hunk 3]
```diff
@@ -225,53 +229,73 @@ function right (msg) {
...
-        resolve(line);
-        return false;
-      }
-      i++;
+    lineReader.eachLine(diaperFileName, (line) => {
+      diapersInMemory.push(line.split('\t'));
+    }, (err) => {
+      if (err) { return reject(err); };
+
+      // Remove header from array (removes first item)
```
Comment (apkatsikas) [inline app.js:286]: @rmag42 here's where I remove the header

File: pantheon.tsv [hunk 4]
```diff
@@ -11340,12 +11340,12 @@ en_curid	name	numlangs	birthcity	birthstate	countryName	countryCode	countryCode3
...
 38646961	Eric Hobsbawm	49	Alexandria		Egypt	EG	EGY	31.2	29.916667	Africa	1917	Male	HISTORIAN	HISTORY	HUMANITIES	2557682	7.711119619	40990.92032	952975	1604707	52197.59184	23.37225664
 38707829	Nikolai Chernyshevsky	33	Saratov		Russia	RU	RUS	51.533333	46.016667	Europe	1828	Male	WRITER	LANGUAGE	HUMANITIES	615101	6.393446529	4589.213995	86510	528591	18639.42424	22.91685363
-38707829	Mat Lundin	33	Austin		Frogworld	RU	RUS	51.533333	46.016667	Europe	1828	Male	UNEMPLOYED	LANGUAGE	HUMANITIES	615101	6.393446529	4589.213995	86510	528591	18639.42424	22.91685363
-38707829	Ryan Agnew	33	Saratov		Gansett Shore	RU	RUS	51.533333	46.016667	Europe	1828	Male	PICKER	LANGUAGE	HUMANITIES	615101	6.393446529	4589.213995	86510	528591	18639.42424	22.91685363
-38707829	Drew Katsikas	33	Saratov		Lizard Land	RU	RUS	51.533333	46.016667	Europe	1828	Male	KATSMARCUS	LANGUAGE	HUMANITIES	615101	6.393446529	4589.213995	86510	528591	18639.42424	22.91685363
-38707829	Jake Walker	33	Saratov		Couch	RU	RUS	51.533333	46.016667	Europe	1828	Male    SCOOPER	LANGUAGE	HUMANITIES	615101	6.393446529	4589.213995	86510	528591	18639.42424	22.91685363
```
Comment (apkatsikas) [inline pantheon.tsv:11346]: jake's formatting was a bit mangled, missing a tab

File: app.js [hunk 5]
```diff
@@ -119,19 +126,19 @@ const discordFunctions = [
...
     fn: pickADiaper,
-    doc: '- poses the question: which famous person\'s full poopy diaper will you choose to smell?'
+    doc: '+ {optional argument \'popular\'} poses the question: which famous person\'s full poopy diaper will you choose to smell?'
   }
 ];
 
 // Help menu so we can see the available commands from Discord
 function help (msg) {
   const channelId = msg.channel.id;
-  const quotes = /['"]+/g;
```
Comment (apkatsikas) [inline app.js:129]: @mat-lundin I changed this regex to only do double quotes. it was stripping the apostrophe/single quotes we want in it

**Pull request merged**